app/code/Magento/ProductAlertGraphQl/Model/Resolver/Price/Unsubscribe.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\ProductAlertGraphQl\Model\Resolver\Price;
+
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\ProductAlert\Helper\Data as AlertsHelper;
+use Magento\ProductAlert\Model\PriceFactory;
+
+/**
+ * Unsubscribe to product price alert
+ */
+class Unsubscribe implements ResolverInterface
+{
+    /**
+     * @var AlertsHelper
+     */
+    private $helper;
+
+    /**
+     * @var PriceFactory
+     */
+    private $priceFactory;
+
+    /**
+     * @param AlertsHelper $helper
+     * @param PriceFactory $priceFactory
+     */
+    public function __construct(
+        AlertsHelper $helper,
+        PriceFactory $priceFactory
+    ) {
+        $this->helper = $helper;
+        $this->priceFactory = $priceFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
+    {
+        if (!$this->helper->isPriceAlertAllowed()) {
+            throw new GraphQlInputException(__('The product price alerts is currently disabled.'));
+        }
+
+        $customerId = $context->getUserId();
+        $store = $context->getExtensionAttributes()->getStore();
+
+        /* Guest checking */
+        if (null === $customerId || 0 === $customerId) {
```

Comment:
This can be simplified to
```
if (!$customerId) {
```
Also, please be consistent with translated sentences ending with .